libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -1009,6 +1009,9 @@ bool WiFiClientSecure::_connectSSL(const char* hostName) {
   _x509_insecure = nullptr;
   _x509_knownkey = nullptr;
 
+  // reduce timeout after successful handshake to fail fast if server stop accepting our data for whathever reason
+  if (ret) _timeout = 5000;
```

Comment:
Seems reasonable here to do this, good catch.

But, given that the object can be reused, it needs to be set at the start of the multiple ::connects and ::connectServers, or on the second pass it will be stuck at the shorter time.

Reply:
I've pushed additional change. All places where TSL context is invalidated calls method _freeSSL. Thus, I've added reset of timeout to 15 seconds there.